src/plugins/ui/evaluation.ts

```diff
@@ -223,7 +223,9 @@ export class EvaluationPlugin extends UIPlugin {
      */
     function _range(range: Range): any[][] {
       const sheet = sheets[range.sheetId]!;
-
+      if (range.invalidSheetName) {
```

Comment:
The real cause of the bug that this PR fixes is the exclamation point at the line above. It's false because we are not sure here that the sheet actually exist. Time has shown that mostly every exclamation point is an hidden bug. An exclamation point can be use for exemple after a check:
```ts
assertSheetValid(range.sheetId);
const sheet = sheets[range.sheetId]!;
```
and, even better, removed:
```ts
assertSheetValid(range.sheetId);
const sheet = sheets[range.sheetId];
if (!sheet) {
  throw new Error("invalid sheet"); // The text does not need to be translated, if the code fall here, it's a programming error as the line assertSheetValid should have thrown
}
```

Now, regarding the issue, we can see that, in the two functions concerned (`refFn` and `_range`), the same checks are done:
- check that the sheet name is correct
- check that the zone is valid

So, I suggest to create an helper (`assertRangeValid`) which would make the two checks.
```ts
function assertRangeValid(range: Range): void {
  if (range.invalidSheetName) {
    throw new Error(...);
  }
  if (!isZoneValid(range.zone)) {
    throw new Error(...);
  }
}

function _range(range: Range): any[][] {
  assertRangeValid(range);
  const sheet = sheets[range.sheetId];
  if (!sheet) {
    throw new Error(...);
  }
  ...
}

function refFn(...) {
  const range: Range = references[position];
  assertRangeValid(range);
  ...
}
```

Please note that a test of `ANYEXPECTED` will have to be adapted, you can simply change the name of the sheet to `sheet1`